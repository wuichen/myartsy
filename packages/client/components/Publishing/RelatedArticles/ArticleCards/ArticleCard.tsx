import { Box, color, Flex, Sans, Serif } from "@artsy/palette"
import { compact } from "lodash"
import React, { Component } from "react"
import track, { TrackingProp } from "react-tracking"
import styled from "styled-components"

import { media } from "Components/Helpers"
import { Byline } from "Components/Publishing/Byline/Byline"
import { Date as DateComponent } from "Components/Publishing/Byline/Date"
import { formatTime, getMediaDate } from "Components/Publishing/Constants"
import { IconVideoPlay } from "Components/Publishing/Icon/IconVideoPlay"
import { crop } from "Utils/resizer"

interface Props {
  article?: any
  color?: string
  editing?: boolean
  editDate?: any
  editDescription?: any
  editTitle?: any
  editImage?: any
  series?: any
  tracking?: TrackingProp
}

@track()
export class ArticleCard extends Component<Props> {
  public static defaultProps: Partial<Props>

  isUnpublishedMedia = () => {
    const { article } = this.props
    return (
      article.layout === "video" && article.media && !article.media.published
    )
  }

  isEditing = () => {
    const {
      editDate,
      editDescription,
      editImage,
      editTitle,
      editing,
    } = this.props

    return editing || editDate || editDescription || editImage || editTitle
  }

  renderDate = () => {
    const { article, editDate } = this.props
    const hasMedia =
      article.media && compact(Object.values(article.media)).length

    if (editDate) {
      return (
        <Sans size="3t" weight="medium">
          {editDate}
        </Sans>
      )
    } else if (hasMedia) {
      return this.renderMediaDate()
    } else {
      return (
        <Byline
          article={article}
          color={this.props.color}
          size="3t"
          layout="condensed"
        />
      )
    }
  }

  renderMediaDate = () => {
    const { article } = this.props
    const mediaDate = getMediaDate(article)
    const date = article.layout === "video" ? mediaDate : article.published_at
    const formatedDate = new Date(date).toISOString()

    if (this.isUnpublishedMedia()) {
      return (
        <Sans size="3t" weight="medium">
          <Flex alignItems="flex-end">
            <Box mr="5px">Available</Box>
            <DateComponent format="monthYear" date={mediaDate} />
          </Flex>
        </Sans>
      )
    } else {
      return <DateComponent layout="condensed" size="3t" date={formatedDate} />
    }
  }

  renderMediaCoverInfo = () => {
    const { article } = this.props

    if (this.isUnpublishedMedia()) {
      return (
        <MediaContainer>
          <Sans size={["8", "8", "10", "10"]}>Coming Soon</Sans>
        </MediaContainer>
      )
    } else {
      return (
        <MediaContainer justifyContent="space-between" alignItems="flex-end">
          <IconVideoPlay color={this.props.color} />
          <Sans size="3t" weight="medium" lineHeight="1em">
            {formatTime(article.media.duration)}
          </Sans>
        </MediaContainer>
      )
    }
  }

  openLink = e => {
    e.preventDefault()

    if (!this.isUnpublishedMedia() && !this.isEditing()) {
      window.open(e.currentTarget.attributes.href.value)
    }

    this.props.tracking.trackEvent({
      action: "Click",
      label: "Related article card",
    })
  }

  render() {
    const {
      article,
      editDescription,
      editImage,
      editTitle,
      series,
    } = this.props
    const { layout } = article
    const isUnpublishedMedia = this.isUnpublishedMedia()

    const description = editDescription ? editDescription : article.description

    const title = editTitle
      ? editTitle
      : article.thumbnail_title || article.title

    return (
      <ArticleCardContainer
        href={isUnpublishedMedia ? "" : article.slug}
        color={this.props.color}
        className="ArticleCard"
        published={!isUnpublishedMedia}
        onClick={this.openLink}
      >
        <Flex
          flexDirection={["column-reverse", "column-reverse", "row", "row"]}
          p={[20, 20, 30, 30]}
        >
          <Flex
            width={["100%", "100%", "50%", "50%"]}
            mb={[0, 0, "5px", "5px"]}
            flexDirection="column"
            justifyContent="space-between"
          >
            <div>
              {series && (
                <Sans size="3t" weight="medium" mb={1}>
                  {series.title}
                </Sans>
              )}
              <Sans size={["8", "8", "10", "10"]} mb={20}>
                {title}
              </Sans>

              <Serif size={["4", "4", "5", "5"]} mb={20}>
                {description}
              </Serif>
            </div>
            {this.renderDate()}
          </Flex>

          <ArticleCardImageContainer
            width={["100%", "100%", "50%", "50%"]}
            ml={[0, 0, 30, 30]}
            mb={[10, 10, 0, 0]}
          >
            {editImage ? (
              editImage
            ) : (
              <Image
                src={crop(article.thumbnail_image, { width: 680, height: 450 })}
              />
            )}
            {article.media && layout === "video" && this.renderMediaCoverInfo()}
          </ArticleCardImageContainer>
        </Flex>
      </ArticleCardContainer>
    )
  }
}

ArticleCard.defaultProps = {
  color: color("black100"),
}

const Image = styled.img`
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
  transition: opacity 0.15s;
  display: block;
`

export const ArticleCardImageContainer = styled(Box)`
  position: relative;
  height: inherit;
  background: white;
`

export const ArticleCardContainer = styled.a<{ published: boolean }>`
  border: 1px solid;
  border-radius: 2px;
  color: ${props => props.color};
  cursor: ${props => (props.published ? "pointer" : "default")};
  text-decoration: none;
  display: block;

  ${Image} {
    opacity: ${props => (props.published ? "1" : "0.7")};
  }

  &:hover {
    ${Image} {
      opacity: 0.7;
    }
  }
`

const MediaContainer = styled(Flex)`
  position: absolute;
  left: 20px;
  right: 20px;
  bottom: 20px;

  svg {
    height: 40px;
  }
  ${media.md`
    svg {
      height: 30px;
    }
  `};
`
