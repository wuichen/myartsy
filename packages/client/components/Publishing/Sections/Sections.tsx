import { Box } from "@artsy/palette"
import { targetingData } from "Components/Publishing/Display/DisplayTargeting"
import { getSlideshowImagesFromArticle } from "Components/Publishing/Layouts/ArticleWithFullScreen"
import { AdDimension, AdUnit } from "Components/Publishing/Typings"
import { clone, compact, findLastIndex, get, once } from "lodash"
import React, { Component } from "react"
import ReactDOM from "react-dom"
import styled from "styled-components"
import { pMedia } from "../../Helpers"
import { DisplayAd } from "../Display/DisplayAd"
import { ArticleData } from "../Typings"
import { Authors } from "./Authors"
import { Embed } from "./Embed"
import { ImageCollection } from "./ImageCollection"
import { ImageSetPreview } from "./ImageSetPreview"
import { SectionContainer } from "./SectionContainer"
import { SocialEmbed } from "./SocialEmbed"
import { Text } from "./Text"
import { Video } from "./Video"

interface Props {
  DisplayPanel?: any
  article: ArticleData
  color?: string
  isMobile?: boolean
  showTooltips?: boolean
  isSponsored?: boolean
  isSuper?: boolean
  customWidth?: number
  isTruncatedAt?: number
  hideAds?: boolean
}

interface State {
  shouldInjectMobileDisplay: boolean
}

/**
 * When isMobile, hide sidebar and inject DisplayAd into the body of the
 * article at a specific paragraph index.
 */
const MOBILE_DISPLAY_INJECT_INDEX = 1
const MOBILE_DISPLAY_INJECT_ID_PREFIX = "__mobile_display_inject__"

export class Sections extends Component<Props, State> {
  static defaultProps = {
    isMobile: false,
  }

  displayInjectId: string

  state = {
    shouldInjectMobileDisplay: false,
  }

  UNSAFE_componentWillMount() {
    const {
      article: { layout },
      isMobile,
    } = this.props

    this.injectDisplayPanelMarker = once(this.injectDisplayPanelMarker)
    const shouldInjectMobileDisplay = isMobile && layout !== "feature"

    this.setState({
      shouldInjectMobileDisplay,
    })
  }

  componentDidMount() {
    if (this.state.shouldInjectMobileDisplay) {
      this.mountDisplayToMarker()
    }
  }

  componentDidUpdate(prevProps) {
    const { isMobile } = this.props

    if (prevProps.isMobile !== isMobile) {
      this.setState(
        {
          shouldInjectMobileDisplay: isMobile,
        },
        () => {
          if (isMobile && this.state.shouldInjectMobileDisplay) {
            this.mountDisplayToMarker()
          }
        }
      )
    }
  }

  /**
   * Inject DisplayAd after a specific paragraph index
   */
  injectDisplayPanelMarker(body) {
    const articleId = this.props.article.id
    const tag = "</p>"
    const updatedBody = compact(body.split(tag))
      .map(p => p + tag)
      .reduce((arr, block, paragraphIndex) => {
        if (paragraphIndex === MOBILE_DISPLAY_INJECT_INDEX) {
          this.displayInjectId = `${MOBILE_DISPLAY_INJECT_ID_PREFIX}${articleId}`
          return arr.concat([block, `<div id="${this.displayInjectId}"></div>`])
        } else {
          return arr.concat([block])
        }
      }, [])
      .join("")

    return updatedBody
  }

  mountDisplayToMarker() {
    const { DisplayPanel } = this.props
    const displayMountPoint = document.getElementById(this.displayInjectId)

    if (displayMountPoint) {
      ReactDOM.render(<DisplayPanel />, displayMountPoint)
    } else {
      console.error(
        "(reaction/Sections.tsx) Error mounting Display: DOM node ",
        "not found",
        displayMountPoint
      )
    }
  }

  getContentStartIndex = () => {
    const {
      article: { layout, sections },
    } = this.props

    if (layout === "feature") {
      const firstText = sections.findIndex(section => {
        return section.type === "text"
      })
      return firstText
    }
  }

  getContentEndIndex = () => {
    const {
      article: { layout, sections },
    } = this.props

    if (["feature", "standard"].includes(layout)) {
      const lastText = findLastIndex(sections, section => {
        return section.type === "text"
      })
      return lastText
    }
  }

  getSection(section, index) {
    const { article, color, customWidth, isMobile, showTooltips } = this.props
    const targetHeight =
      customWidth && (customWidth > 750 && !isMobile) ? 750 : 500
    const size = customWidth && { width: customWidth }
    const fullscreenImages = getSlideshowImagesFromArticle(article)

    const sections = {
      image_collection: (
        <ImageCollection
          color={color}
          sectionLayout={section.layout}
          articleLayout={article.layout}
          images={section.images}
          targetHeight={targetHeight}
          gutter={10}
          size={size}
          fullscreenImages={fullscreenImages}
        />
      ),
      image_set: (
        <ImageSetPreview
          section={section}
          color={color}
          fullscreenImages={fullscreenImages}
        />
      ),
      video: <Video section={section} color={color} />,
      embed: <Embed section={section} />,
      social_embed: <SocialEmbed section={section} />,
      text: (
        <Text
          color={color}
          html={section.body}
          layout={article.layout}
          isContentStart={index === this.getContentStartIndex()}
          isContentEnd={index === this.getContentEndIndex()}
          showTooltips={showTooltips}
        />
      ),
      default: false,
    }

    const sectionComponent = sections[section.type] || sections.default
    return sectionComponent
  }

  getAdUnit(index: number, indexAtFirstAd: number) {
    const { isMobile } = this.props

    if (index === indexAtFirstAd) {
      return isMobile
        ? AdUnit.Mobile_Feature_InContentLeaderboard1
        : AdUnit.Desktop_Feature_Leaderboard1
    }

    if (index === 6) {
      return isMobile
        ? AdUnit.Mobile_Feature_InContentLeaderboard2
        : AdUnit.Desktop_Feature_Leaderboard2
    }

    return isMobile
      ? AdUnit.Mobile_Feature_InContentLeaderboard3
      : AdUnit.Desktop_Feature_LeaderboardRepeat
  }

  /**
   * Add a margin top when to the ad when the section is any of the following:
   * 1) Text section without <br> tags wrapping the <p> or <h2> content
   * 3) Image set section
   * 4) Image collection with a caption/title
   */
  getAdMarginTop(section) {
    const shouldAddTopMargin = () => {
      switch (section.type) {
        case "image_set":
          return true
        case "image_collection":
          get(section, ["images", "[0]", "caption"])
          return true
        case "text":
          section.body.includes("<br>")
          return true
        default:
          return false
      }
    }

    return shouldAddTopMargin() ? "calc(100% - 96%)" : null
  }

  renderSections() {
    const {
      article,
      customWidth,
      isMobile,
      isSponsored,
      isSuper,
      isTruncatedAt,
      hideAds,
    } = this.props
    const { shouldInjectMobileDisplay } = this.state
    let quantityOfAdsRendered = 0
    let firstAdInjected = false
    let placementCount = 1
    let displayMarkerInjected = false
    let indexAtFirstAd = null

    const articleSections = isTruncatedAt
      ? clone(article.sections).slice(0, isTruncatedAt)
      : article.sections

    const renderedSections = articleSections.map((sectionItem, index) => {
      let section = sectionItem
      let ad = null
      const shouldInject =
        shouldInjectMobileDisplay &&
        sectionItem.type === "text" &&
        !displayMarkerInjected

      /**
       *  Possible data types for sectionItem.type are
       *  callout, embed, social_embed, text, slideshow, image_set, and image_collection.
       *  Here we want to inject the first ad after the first image_collection OR image_set data type.
       * */
      let shouldInjectNewAds =
        article.layout === "feature" &&
        (sectionItem.type === "image_collection" ||
          sectionItem.type === "image_set") &&
        !firstAdInjected &&
        !isSuper &&
        !hideAds

      if (firstAdInjected) {
        placementCount++
      }

      if (placementCount % 6 === 0 && quantityOfAdsRendered < 2) {
        // only render 2 ads on Features
        shouldInjectNewAds = true
      }

      if (shouldInjectNewAds) {
        quantityOfAdsRendered++

        const adDimension = isMobile
          ? isSponsored
            ? AdDimension.Mobile_Sponsored_Feature_InContentLeaderboard1
            : AdDimension.Mobile_Feature_InContentLeaderboard1
          : AdDimension.Desktop_NewsLanding_Leaderboard1

        const marginTop = this.getAdMarginTop(section)

        firstAdInjected = true
        indexAtFirstAd = indexAtFirstAd === null ? index : indexAtFirstAd // only set this value once; after the index where 1st ad injection is found

        ad = (
          <AdWrapper mt={marginTop}>
            <DisplayAd
              adUnit={this.getAdUnit(placementCount, indexAtFirstAd)}
              adDimension={adDimension}
              targetingData={targetingData(
                article,
                isSponsored ? "sponsorfeature" : "feature"
              )}
            />
          </AdWrapper>
        )
      }

      if (shouldInject) {
        try {
          section = clone(sectionItem)
          section.body = this.injectDisplayPanelMarker(section.body)
          displayMarkerInjected = true
        } catch (error) {
          console.error(
            "(reaction/Sections.jsx) Error injecting Display:",
            error
          )
        }
      }

      const child = this.getSection(section, index)

      if (child) {
        return (
          <SectionContainer
            key={index}
            articleLayout={article.layout}
            section={section}
            customWidth={customWidth}
          >
            {child}
            {ad}
          </SectionContainer>
        )
      }
    })

    return renderedSections
  }

  renderAuthors() {
    const {
      color,
      article: { authors },
    } = this.props

    if (authors) {
      return (
        <SectionContainer>
          <Authors authors={authors} color={color} />
        </SectionContainer>
      )
    }
  }

  renderPostScript() {
    const { article, color } = this.props
    const { layout, postscript } = article

    if (postscript) {
      return (
        <SectionContainer>
          <Text
            color={color}
            html={postscript}
            layout={layout}
            postscript={Boolean(postscript)}
          />
        </SectionContainer>
      )
    }
  }

  render() {
    const { article } = this.props

    return (
      <StyledSections layout={article.layout}>
        {this.renderSections()}
        {this.renderAuthors()}
        {this.renderPostScript()}
      </StyledSections>
    )
  }
}

const chooseMargin = layout => {
  if (layout === "standard") {
    return "0;"
  } else if (layout === "feature") {
    return "80px auto 0 auto;"
  }
}

export const StyledSections = styled.div<{ layout: string }>`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  margin: ${({ layout }) => chooseMargin(layout)};
  max-width: ${({ layout }) => (layout === "standard" ? "780px" : "auto")};

  ${({ layout }) => pMedia.xl`
    max-width: ${layout === "standard" ? "680px" : "auto"};
    ${layout === "feature" ? "margin: 80px auto 0 auto" : ""}
  `}

  ${({ layout }) => pMedia.md`
    max-width: ${layout === "standard" ? "780px" : "auto"};
  `}

  ${({ layout }) => pMedia.xs`
    max-width: ${layout === "standard" ? "780px" : "auto"};
    ${layout === "feature" ? "margin: 30px auto 0 auto" : ""}
  `}
`
const AdWrapper = styled(Box)`
  width: 100vw;
  margin-left: calc(-50vw + 50%);
`
