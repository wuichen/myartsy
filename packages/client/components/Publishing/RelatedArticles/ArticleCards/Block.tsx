import { Box, color } from "@artsy/palette"
import React from "react"
import styled from "styled-components"

import { unica } from "Assets/Fonts"
import {
  Vertical,
  VerticalOrSeriesTitle,
} from "Components/Publishing/Sections/VerticalOrSeriesTitle"
import { SeriesAbout } from "Components/Publishing/Series/SeriesAbout"
import { ArticleData } from "Components/Publishing/Typings"
import { ArticleCards } from "./ArticleCards"

interface Props {
  article?: ArticleData
  color?: string
  relatedArticles?: any
}

export const ArticleCardsBlock: React.SFC<Props> = props => {
  const { article, relatedArticles } = props
  const { seriesArticle } = article

  return (
    <ArticleCardsContainer color={props.color}>
      {(relatedArticles || article.relatedArticles) && (
        <Box maxWidth={1200} mx="auto">
          <VerticalOrSeriesTitle
            article={article}
            color={props.color}
            prefix="More in "
          />
          <ArticleCards
            relatedArticles={relatedArticles || article.relatedArticles}
            series={seriesArticle}
            color={props.color}
          />
        </Box>
      )}
      {seriesArticle && (
        <Box maxWidth={1200} mx="auto" pt={[40, 40, 60]} pb={100}>
          <SeriesAbout article={seriesArticle} color={props.color} />
        </Box>
      )}
    </ArticleCardsContainer>
  )
}

ArticleCardsBlock.defaultProps = {
  color: color("black100"),
}

export const ArticleCardsContainer = styled.div`
  color: ${props => props.color};

  ${Vertical} {
    ${unica("s32")};
    width: 100%;
    margin-bottom: 40px;

    a {
      border-bottom: 2px solid;
    }
  }
`
