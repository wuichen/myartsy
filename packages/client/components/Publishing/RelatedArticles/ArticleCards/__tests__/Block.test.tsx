import {
  FeatureArticle,
  SeriesArticleSponsored,
  StandardArticle,
  VideoArticle,
} from "Components/Publishing/Fixtures/Articles"
import { VerticalOrSeriesTitle } from "Components/Publishing/Sections/VerticalOrSeriesTitle"
import { SeriesAbout } from "Components/Publishing/Series/SeriesAbout"
import { mount } from "enzyme"
import "jest-styled-components"
import { clone } from "lodash"
import React from "react"
import renderer from "react-test-renderer"
import { ArticleCard } from "../ArticleCard"
import { ArticleCardsBlock } from "../Block"

describe("ArticleCard", () => {
  it("renders properly", () => {
    const component = renderer
      .create(
        <ArticleCardsBlock
          article={StandardArticle}
          relatedArticles={[StandardArticle, VideoArticle]}
        />
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })

  it("renders a list of articles properly", () => {
    const component = mount(
      <ArticleCardsBlock
        article={StandardArticle}
        relatedArticles={[StandardArticle, VideoArticle]}
      />
    )
    expect(component.text()).toMatch(StandardArticle.vertical.name)
    expect(component.find(ArticleCard).length).toBe(2)
  })

  it("renders a list of articles in a series properly", () => {
    const FeatureSeriesArticle = clone({
      ...FeatureArticle,
      seriesArticle: SeriesArticleSponsored,
    })
    const component = mount(
      <ArticleCardsBlock
        article={FeatureSeriesArticle}
        relatedArticles={[StandardArticle, VideoArticle]}
      />
    )
    expect(component.find(VerticalOrSeriesTitle).length).toBe(1)
    expect(component.find(VerticalOrSeriesTitle).text()).toMatch(
      SeriesArticleSponsored.title
    )
    expect(component.find(ArticleCard).length).toBe(2)
    expect(component.find(SeriesAbout).length).toBe(1)
  })
})
