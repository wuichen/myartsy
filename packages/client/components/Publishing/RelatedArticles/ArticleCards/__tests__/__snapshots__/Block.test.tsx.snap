// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard renders properly 1`] = `
.c1 {
  max-width: 1200px;
  margin-left: auto;
  margin-right: auto;
}

.c5 {
  color: #000;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  padding: 20px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 0px;
  width: 100%;
}

.c4 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 24px;
  color: #000;
}

.c10 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 28px;
  line-height: 36px;
  margin-bottom: 20px;
}

.c11 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 18px;
  line-height: 26px;
  margin-bottom: 20px;
}

.c13 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
}

.c21 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 1em;
}

.c3 a {
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c14 {
  margin: 10px 20px 0 0;
}

.c14::before {
  content: "";
  display: inline-block;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin: 0 5px 1px 0;
  background-color: #000;
}

.c15 {
  margin: 5px 20px 0 0;
  white-space: nowrap;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  color: #000;
}

.c20 {
  width: 32px;
  height: 32px;
}

.c18 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
  -webkit-transition: opacity 0.15s;
  transition: opacity 0.15s;
  display: block;
}

.c16 {
  margin-bottom: 10px;
  margin-left: 0px;
  width: 100%;
  position: relative;
  height: inherit;
  background: white;
}

.c7 {
  border: 1px solid;
  border-radius: 2px;
  color: #000;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
}

.c7 .c17 {
  opacity: 1;
}

.c7:hover .c17 {
  opacity: 0.7;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: absolute;
  left: 20px;
  right: 20px;
  bottom: 20px;
}

.c19 svg {
  height: 40px;
}

.c6 {
  margin-bottom: 40px;
}

.c0 {
  color: #000;
}

.c0 .c2 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 32px;
  line-height: 1.1em;
  width: 100%;
  margin-bottom: 40px;
}

.c0 .c2 a {
  border-bottom: 2px solid;
}

@media screen and (min-width:40em) {
  .c8 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media screen and (min-width:52em) {
  .c8 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:64em) {
  .c8 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:40em) {
  .c8 {
    padding: 20px;
  }
}

@media screen and (min-width:52em) {
  .c8 {
    padding: 30px;
  }
}

@media screen and (min-width:64em) {
  .c8 {
    padding: 30px;
  }
}

@media screen and (min-width:40em) {
  .c9 {
    margin-bottom: 0px;
  }
}

@media screen and (min-width:52em) {
  .c9 {
    margin-bottom: 5px;
  }
}

@media screen and (min-width:64em) {
  .c9 {
    margin-bottom: 5px;
  }
}

@media screen and (min-width:40em) {
  .c9 {
    width: 100%;
  }
}

@media screen and (min-width:52em) {
  .c9 {
    width: 50%;
  }
}

@media screen and (min-width:64em) {
  .c9 {
    width: 50%;
  }
}

@media screen and (min-width:40em) {
  .c10 {
    font-size: 28px;
  }
}

@media screen and (min-width:52em) {
  .c10 {
    font-size: 42px;
  }
}

@media screen and (min-width:64em) {
  .c10 {
    font-size: 42px;
  }
}

@media screen and (min-width:40em) {
  .c10 {
    line-height: 36px;
  }
}

@media screen and (min-width:52em) {
  .c10 {
    line-height: 50px;
  }
}

@media screen and (min-width:64em) {
  .c10 {
    line-height: 50px;
  }
}

@media screen and (min-width:40em) {
  .c11 {
    font-size: 18px;
  }
}

@media screen and (min-width:52em) {
  .c11 {
    font-size: 22px;
  }
}

@media screen and (min-width:64em) {
  .c11 {
    font-size: 22px;
  }
}

@media screen and (min-width:40em) {
  .c11 {
    line-height: 26px;
  }
}

@media screen and (min-width:52em) {
  .c11 {
    line-height: 32px;
  }
}

@media screen and (min-width:64em) {
  .c11 {
    line-height: 32px;
  }
}

@media screen and (min-width:40em) {
  .c16 {
    margin-bottom: 10px;
    margin-left: 0px;
  }
}

@media screen and (min-width:52em) {
  .c16 {
    margin-bottom: 0px;
    margin-left: 30px;
  }
}

@media screen and (min-width:64em) {
  .c16 {
    margin-bottom: 0px;
    margin-left: 30px;
  }
}

@media screen and (min-width:40em) {
  .c16 {
    width: 100%;
  }
}

@media screen and (min-width:52em) {
  .c16 {
    width: 50%;
  }
}

@media screen and (min-width:64em) {
  .c16 {
    width: 50%;
  }
}

@media (max-width:900px) {
  .c19 svg {
    height: 30px;
  }
}

@media screen and (min-width:40em) {
  .c6 {
    margin-bottom: 60px;
  }
}

<div
  className="c0"
  color="#000"
>
  <div
    className="c1"
  >
    <div
      className="c2 c3 c4"
      color="#000"
      fontFamily={
        Object {
          "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
          "fontWeight": 500,
        }
      }
      fontSize="14px"
    >
      More in 
      <span>
        Art Market
      </span>
    </div>
    <div
      className="c5"
      color="#000"
    >
      <div
        className="c6"
      >
        <a
          className="ArticleCard c7"
          color="#000"
          href="new-yorks-next-art-district"
          onClick={[Function]}
        >
          <div
            className="c8"
          >
            <div
              className="c9"
              width={
                Array [
                  "100%",
                  "100%",
                  "50%",
                  "50%",
                ]
              }
            >
              <div>
                <div
                  className="c10"
                  fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
                  fontSize={
                    Array [
                      "28px",
                      "28px",
                      "42px",
                      "42px",
                    ]
                  }
                >
                  New York's Next Art District
                </div>
                <div
                  className="c11"
                  fontFamily="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
                  fontSize={
                    Array [
                      "18px",
                      "18px",
                      "22px",
                      "22px",
                    ]
                  }
                >
                  Land exhibitions, make influential contacts, and gain valuable feedback about your work.
                </div>
              </div>
              <div
                className="Byline c12"
                color="#000"
              >
                <div
                  className="c13"
                  fontFamily={
                    Object {
                      "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                      "fontWeight": 500,
                    }
                  }
                  fontSize="14px"
                >
                  <div
                    className="c14"
                    color="#000"
                  >
                    Casey Lesser
                  </div>
                </div>
                <div
                  className="c15"
                >
                  <div
                    className="c13"
                    fontFamily={
                      Object {
                        "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                        "fontWeight": 500,
                      }
                    }
                    fontSize="14px"
                  >
                    May 19, 2017
                  </div>
                </div>
              </div>
            </div>
            <div
              className="c16"
              width={
                Array [
                  "100%",
                  "100%",
                  "50%",
                  "50%",
                ]
              }
            >
              <img
                className="c17 c18"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2F7lsxxsw0qPAuKl37jEYitw%252Farticle%2Basset%2B1-hig%2Bres%2Bcopy.jpg&width=680&height=450&quality=95"
              />
            </div>
          </div>
        </a>
      </div>
      <div
        className="c6"
      >
        <a
          className="ArticleCard c7"
          color="#000"
          href="joanne-artman-gallery-poetry-naturerefinement-form"
          onClick={[Function]}
        >
          <div
            className="c8"
          >
            <div
              className="c9"
              width={
                Array [
                  "100%",
                  "100%",
                  "50%",
                  "50%",
                ]
              }
            >
              <div>
                <div
                  className="c10"
                  fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
                  fontSize={
                    Array [
                      "28px",
                      "28px",
                      "42px",
                      "42px",
                    ]
                  }
                >
                  New Study Shows the Gender Pay Gap for Artists Is Not So Simple
                </div>
                <div
                  className="c11"
                  fontFamily="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
                  fontSize={
                    Array [
                      "18px",
                      "18px",
                      "22px",
                      "22px",
                    ]
                  }
                >
                  The elegant spiral of the Nautilus shell, the sinuous pattern of the banks of a river, or the swirling vortex street of clouds - patterns exist.
                </div>
              </div>
              <div
                className="c15"
              >
                <div
                  className="c13"
                  fontFamily={
                    Object {
                      "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                      "fontWeight": 500,
                    }
                  }
                  fontSize="14px"
                >
                  Aug 28, 2017
                </div>
              </div>
            </div>
            <div
              className="c16"
              width={
                Array [
                  "100%",
                  "100%",
                  "50%",
                  "50%",
                ]
              }
            >
              <img
                className="c17 c18"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2FwHFgQlrTrHav5O6bQRJ0dg%252FUntitled%2BSuspended_30x67x33%2B%25282%2529_sm%2Bcropped.jpg&width=680&height=450&quality=95"
              />
              <div
                className="c19"
              >
                <svg
                  className="c20"
                  version="1.1"
                  viewBox="0 0 40 56"
                  x="0px"
                  xmlns="http://www.w3.org/2000/svg"
                  y="0px"
                >
                  <g
                    fill="none"
                    fillRule="evenodd"
                    stroke="none"
                    strokeWidth="1"
                  >
                    <polygon
                      fill="#000"
                      points="0 0 0 56 39.3600006 28"
                    />
                  </g>
                </svg>
                <div
                  className="c21"
                  fontFamily={
                    Object {
                      "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                      "fontWeight": 500,
                    }
                  }
                  fontSize="14px"
                >
                  02:28
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
